FinSearchUnified/Bundle/SearchBundleFindologic/QueryBuilder.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace FinSearchUnified\Bundle\SearchBundleFindologic;
+
+use Exception;
+use Shopware\Bundle\PluginInstallerBundle\Service\InstallerService;
+use Shopware\Components\HttpClient\HttpClientInterface;
+use Shopware\Components\HttpClient\RequestException;
+use Shopware\Components\HttpClient\Response;
+use Shopware_Components_Config;
+
+class QueryBuilder
+{
+    const BASE_URL = 'https://service.findologic.com/ps/xml_2.0/';
+    const ALIVE_ENDPOINT = 'alivetest.php';
+    const SEARCH_ENDPOINT = 'index.php';
+    const NAVIGATION_ENDPOINT = 'selector.php';
+
+    /**
+     * @var HttpClientInterface
+     */
+    protected $httpClient;
+
+    private $parameters;
+
+    private $shopUrl;
+
+    private $shopKey;
+
+    /**
+     * QueryBuilder constructor.
+     *
+     * @param HttpClientInterface $httpClient
+     * @param InstallerService $installerService
+     * @param Shopware_Components_Config $config
+     *
+     * @throws Exception
+     */
+    public function __construct(
+        HttpClientInterface $httpClient,
+        InstallerService $installerService,
+        Shopware_Components_Config $config
+    ) {
+        $plugin = $installerService->getPluginByName('FinSearchUnified');
+
+        $this->parameters = [];
+        $this->httpClient = $httpClient;
+        $this->shopKey = $config->offsetGet('ShopKey');
+        $this->shopUrl = rtrim(Shopware()->Shop()->getHost(), '/') . '/';
+        $this->parameters = [
+            'userip' => $this->getClientIp(),
+            'revision' => $plugin->getVersion(),
+            'shopkey' => $this->shopKey
+        ];
+    }
+
+    /**
+     * @param bool $isSearch
```

Comment:
Also add a short description which makes a statement about the default value e.g. `@param bool $isSearch Defaults to true` would be sufficient.